test/helm_basic_example_template_test.go

```diff
@@ -0,0 +1,120 @@
+// +build kubernetes
+
+// NOTE: we have build tags to differentiate kubernetes tests from non-kubernetes tests. This is done because minikube
+// is heavy and can interfere with docker related tests in terratest. To avoid overloading the system, we run the
+// kubernetes tests separately from the others.
+
+package test
+
+import (
+	"encoding/json"
+	"path/filepath"
+	"testing"
+
+	"github.com/ghodss/yaml"
+	"github.com/stretchr/testify/require"
+	appsv1 "k8s.io/api/apps/v1"
+
+	"github.com/gruntwork-io/terratest/modules/helm"
+)
+
+// This file contains examples of how to use terratest to test helm chart template logic by rendering the templates
+// using `helm template`, and then reading in the rendered templates.
+// There are two tests:
+// - TestHelmBasicExampleTemplateRenderedDeployment: An example of how to read in the rendered object and check the
+//   computed values.
+// - TestHelmBasicExampleTemplateRequiredTemplateArgs: An example of how to check that the required args are indeed
+//   required for the template to render.
+
+// An example of how to verify the rendered template object of a Helm Chart given various inputs.
+func TestHelmBasicExampleTemplateRenderedDeployment(t *testing.T) {
+	t.Parallel()
+
+	// Path to the helm chart we will test
+	helmChartPath, err := filepath.Abs("../examples/helm-basic-example")
+	require.NoError(t, err)
+
+	// Since we aren't deploying any resources, there is no need to setup kubectl authentication, helm home, or
+	// namespaces
+
+	// Setup the args. For this test, we will set the following input values:
+	// - containerImageRepo=nginx
+	// - containerImageTag=1.15.8
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"containerImageRepo": "nginx",
+			"containerImageTag":  "1.15.8",
+		},
+	}
+
+	// Run RenderTemplate to render the template and capture the output. Note that we use the version without `E`, since
+	// we want to assert that the template renders without any errors.
+	// Additionally, although we know there is only one yaml file in the template, we deliberately path a templateFiles
+	// arg to demonstrate how to select individual templates to render.
+	output := helm.RenderTemplate(t, options, helmChartPath, []string{"templates/deployment.yaml"})
+
+	// Now we use kubernetes/client-go library to render the template output into the Deployment struct. This will
+	// ensure the Deployment resource is rendered correctly.
+	// NOTE: the client-go library can only decode json, so we will first convert the yaml to json
+	jsonData, err := yaml.YAMLToJSON([]byte(output))
+	require.NoError(t, err)
+	var deployment appsv1.Deployment
+	require.NoError(t, json.Unmarshal(jsonData, &deployment))
```

Comment:
Is this something you'll need to do regularly when using yaml? If so, should we add a helper method to make this easier? Or even a `RenderTemplateToJson` method?

Reply:
Yes. I actually caught this bug later when I tried using it in `k8s-service`, and then I realized we will do this quite a lot. I think I am thinking along the lines of `UnmarshalK8SYaml`.

Reply:
Ok added an `UnmarshalK8SYaml` function. This section now looks like:

```go
var deployment appsv1.Deployment
helm.UnmarshalK8SYaml(t, out, &deployment)
```